File: 27-Interpreter/src/go/constant.go
package main

// Constant is a Terminal Expression — a literal true/false value.
type Constant struct {
	value bool
}

func NewConstant(value bool) *Constant {
	return &Constant{value: value}
}

func (c *Constant) Interpret(context *Context) bool {
	return c.value
}

File: 27-Interpreter/src/rust/constant.rs
use crate::expression::Expression;
use crate::context::Context;

/// Terminal Expression — a literal true/false constant.
pub struct Constant {
    value: bool,
}

impl Constant {
    pub fn new(value: bool) -> Self {
        Constant { value }
    }
}

impl Expression for Constant {
    fn interpret(&self, _context: &Context) -> bool {
        self.value
    }
}
